src/query/service/src/servers/http/v1/query/http_query.rs

```diff
@@ -139,25 +139,35 @@ pub struct HttpSessionConf {
 }
 
 impl HttpSessionConf {
+    // TODO(liyz): this function is not needed? Client side should apply the affect itself,
+    // and pass the db and settings parameters in the "session" field like:
+    // session: {database: "default", settings: {max_threads: "4"}}.
+    // do not actually need to apply these affect again on the server side.
```

Comment:
with this, client just need to update local session as a whole,  make it  easier.

Reply:
the updating role may get complicated, client code not need to update

Reply:
session by its nature is server side, client side session is a workaound.
only a few change need also apply to client (like database).
so it is better to not add this burden to client.

Reply:
session by its nature is server side.
only a few change need also apply to client (like database).
so it is better to not add this burden to client.

Reply:
> the updating rule may get complicated or change like this pr, client code not need to update
> session by its nature is server side, client side session is a workaound.

no, the client side may also have it's customized settings in their config like:

```
driver = sql.Open("databend://<host>/<db>?max_sessions_limit=1024")
```

this limit might not actually scoped by the sessions' lifetime, but it's considered long as the client's lifetime.

Reply:
I think this is not the common case.

I simple usage, server side do most of the work, client only handle session simply like a blob, no need to know the structure and how to update it.  it also make life much easier when we want to update/fix about the session object..

for advanced clients like you mentioned , you can simply ignore what server return,
just maintain and **construct** the session part as you like and send to server when post query.

Reply:
`QueryAffect` is the (change) event. 

 `Session` is the state, `reply state`  = `req state` [ + `change event`]. it is in fact redundant (since client know `req state` and  `change event`), we introduce it only to make both client and server side easier.

Reply:
I think it's not a good design, if we'd hope to sync what the things have changed during the SQL to the client, we'd better encapsulate it to a single place.

In my previous understanding, QueryAffect plays this role to tell what had changed in the SQL to the client about the local state. But now there're two places to tell the clients what had changed and have different meanings.

In this scenario, I suppose that it'd better to still make QueryAffect to play its role, tells the client the new completed set of session level settings whenever the session had changed.